core/Changes/UserChanges.php

```diff
@@ -42,23 +42,68 @@ public function __construct(array $user, ?Db\AdapterInterface $db = null)
      */
     public function getNewChangesStatus(): int
     {
-        $idchangeLastViewed = (isset($this->user['idchange_last_viewed']) ? $this->user['idchange_last_viewed'] : null);
+        return $this->changesModel->doChangesExist($this->getIdchangeLastViewed());
+    }
+
+    /**
+     * Return the count of new changes unseen by the user
+     *
+     * @return int   Change count
+     * @throws \Exception
+     */
+    public function getNewChangesCount(): int
+    {
+        return $this->changesModel->getNewChangesCount($this->getIdchangeLastViewed());
+    }
 
-        $changesModel = new ChangesModel($this->db);
-        return $changesModel->doChangesExist($idchangeLastViewed);
+    /**
+     * Return the key of the last viewed change for the user, if any
+     *
+     * @return int
+     */
+    private function getIdchangeLastViewed(): ?int
+    {
+        return (isset($this->user['idchange_last_viewed']) ? $this->user['idchange_last_viewed'] : null);
```

Comment:
I don't know much about $this->user regarding type and the class properties don't indicate it either.

How can we guarantee that `$this->user['idchange_last_viewed']` returns a int?

Reply:
`UserManager/Model->getUser()` just returns a fetchAll array of the user db table, so depending on the Db adapter the type might be a string, I'll wrap the return value with `intval()` to make type safety a bit more explicit :+1: